frontend/ui-picasso/app/components/Molecules/AllBondsTable.tsx

```diff
@@ -14,17 +15,46 @@ import { TokenAsset, TokenPairAsset } from "@/components";
 import { BondOffer } from "@/stores/defi/polkadot/bonds/types";
 import { getROI } from "@/defi/polkadot/pallets/BondedFinance";
 import { humanBalance } from "@/utils/formatters";
+import { useQuery } from "@apollo/client";
+import { GET_BONDED_FINANCE } from "@/apollo/queries";
+import BigNumber from "bignumber.js";
 
 export type AllBondsTableProps = TableContainerProps & {
   bonds?: BondOffer[];
   onRowClick: (offerId: string) => void;
 };
 
+function getTotalPurchasedInFormat(
+  currentBond: {
```

Comment:
As we have `if (currentBond) {`, this looks like optional

```suggestion
  currentBond?: {
```